examples/comparison.rs

```diff
@@ -63,7 +63,11 @@ fn print_alloc_info(
 fn main() -> Result<(), Box<dyn Error>> {
     // The string cache is normally lazily initialized which would erroneously show up as a
     // memory leak, so explicitly initialize it here.
+    #[cfg(not(feature = "thread_unsafe"))]
```

Comment:
The feature should not change the API exposed by this crate other than making types Send+Sync or not.

Reply:
I didn't find a good way to do it, how would you do it?

Reply:
There's no need to have differently named string modules - it's not as though you can mix and match a thread safe IString with the thread unsafe IString, unless you duplicated the entire crate API including IValue, since IStrings can be stored inside other IValues. So just leave the API unchanged and use `#[cfg]` to switch implementations of that API.